bon-macros/src/builder/builder_gen/input_func.rs

```diff
@@ -503,3 +504,27 @@ impl Visit<'_> for FindSelfReference {
         }
     }
 }
+
+fn get_must_use_attribute(attrs: &[syn::Attribute]) -> Option<syn::Attribute> {
+    let mut iter = attrs
+        .iter()
+        .filter(|attr| !attr.is_doc() && matches!(attr.style, syn::AttrStyle::Outer))
```

Comment:
I think this entire `filter(...)` here is redundant because the filter below is a subset of the what's checked here. For example `doc` attributes definitely don't have a `must_use` path in them, and nothing but outer attributes are supported on functions.

Reply:
This makes it for a fun reddit/twitter post:
- [Reddit](https://www.reddit.com/r/rust/comments/1f4vvgy/media_fun_fact_you_can_write_functions_docs_in/)
- [Twitter](https://x.com/veetaha/status/1829516618386588060)

Reply:
Without the filter it might also find `must_use` defined as inner attribute? It's still horrible… :D

Reply:
Yeah, we can either filter inner attributes, or produce a compile error if someone uses an inner doc or must_use attribute (they deserve it 👀 )

Reply:
Its horrible, yes, but we probably don't need to filter at all. syn seems to correctly parse them, so why not detect either case and just use it? Its horrible code, but we have one filter less, and it still works. Less complexity on the macro side.

(They still might deserve it :axe:)

Reply:
Yeah, if it doesn't cost us anything in terms of code complexity, let's do that

(But they don't deserve this 💯 )

Reply:
bon adds outer attributes so rust is unhappy about the inner attributes being used. On stable it doesn't work, on nightly its annoyed. No need to create any way around it, it doesn't work in the first place. :D